Extend `@on` decorator to filter arbitrary attributes (#2498)
Textualize/textual: docs/guide/events.md

Author: Still trying to strike a balance between my personal verbose style and the intended style for the docs.
Let me know how this looks.

!!! note

    If multiple decorated handlers match the message, then they will *all* be called in the order they are defined.

    The naming convention handler will be called *after* any decorated handlers.

#### Applying CSS selectors to arbitrary attributes

> The decorator `on` also accepts `**kwargs` that can be used to specify CSS selectors for the specified attributes of the message.

Reviewer: ```suggestion
The `on` decorator also accepts selectors as keyword arguments that may be used to match other attributes in a Message.
```
